peachjam/js/peachjam.ts

```diff
@@ -65,16 +66,24 @@ class PeachJam {
     window.htmx = htmx;
     // htmx:load is fired both when the page loads (weird) and when new content is loaded. We only care about the latter
     // case. See https://github.com/bigskysoftware/htmx/issues/1500
-    const htmxHelper = { firstLoad: true};
-    document.body.addEventListener('htmx:load', (e) => {
+    const htmxHelper = { firstLoad: true };
+    let token: string = '';
+    document.body.addEventListener('htmx:load', async (e) => {
       if (htmxHelper.firstLoad) {
         htmxHelper.firstLoad = false;
+        token = await csrfToken();
```

Comment:
This will load a CSRF token (and hence set the CSRF cookie) on ever page load, which defeats the purpose of doing it via javascript. Instead, we must do this at the last minute when an htmx POST request is about to happen.

Reply:
I've updated this to use the check for the confirm event. Its working okay but it still fires on all post requests. We could check for specific target elements i.e the saveDocument button to make it only work for the saveDocuments functionality